src/Raven.Client/Documents/Session/Operations/LoadOperation.cs

```diff
@@ -169,7 +169,18 @@ public void SetResult(GetDocumentsResult result)
 
             _session.RegisterIncludes(result.Includes);
 
-            _session.RegisterCounters(result.Counters, _counters, _includeAllCounters, _ids);
+            if (_includeAllCounters || _counters != null)
+            {
+                var counterToInclude = new Dictionary<string, string[]>();
+                var counters = _counters ?? new string[0];
+                foreach (var id in _ids)
+                {
+                    counterToInclude[id] = counters;
+                }
+
+                _session.RegisterCounters(result.Counters, counterToInclude);
```

Comment:
I'm not sure that I'm following this code here. Why are we doing this like that? We cannot include different counters per document ID. Seems redundant and a lot of additional work?